### PR TITLE
:seedling: Unit tests for checker result and request

### PR DESCRIPTION
**Pull request by naveensrinivasan**
Included tests for checker result and request

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [x] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: checker/check_result.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ const (
 const (
 	// DetailInfo is info-level log.
 	DetailInfo DetailType = iota
-	// DetailWarn is warn log.
+	// DetailWarn is warned log.
```
Comment (spencerschrock) [inline checker/check_result.go:47]: optional: this one was probably fine as is?

File: .golangci.yml [hunk 2]
```diff
@@ -12,6 +12,13 @@ issues:
   # Default: 3
   max-same-issues: 0
   new-from-rev: ""
+  exclude-rules:
+    - path: '(.+)_test\.go'
+      linters:
+        - funlen
+        - goconst
+        - govet
```
Comment (spencerschrock) [inline .golangci.yml:20]: I understand the other disabled linters, but `govet` can catch useful things in tests.
> Printf calls whose arguments do not align with the format string

I'm sure we get warnings we don't care about, but we will miss bugs that affect test correctness. For example the `loopclosure` functionality in `govet` would catch the data race mentioned in my other comments.

File: checker/check_result_test.go [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+			want: CheckResult{
+				Name:    "",
+				Reason:  "runtime error",
+				Score:   -1,
+				Version: 2,
+				Error:   errors.New("runtime error"), //nolint:goerr113
+			},
+		},
+	}
+	for _, tt := range tests {
```
Comment (spencerschrock) [inline checker/check_result_test.go:465]: please shadow tt here to avoid race conditions
```
tt := tt
```

File: checker/check_result_test.go [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+			},
+			want: CheckResult{
+				Name:    "name",
+				Reason:  "reason",
+				Score:   -1,
+				Version: 2,
+			},
+		},
+	}
+	for _, tt := range tests {
```
Comment (spencerschrock) [inline checker/check_result_test.go:429]: please shadow tt here to avoid race conditions
```
tt := tt
```

File: checker/check_result_test.go [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+			},
+			want: CheckResult{
+				Name:    "name",
+				Reason:  "reason",
+				Score:   0,
+				Version: 2,
+			},
+		},
+	}
+	for _, tt := range tests {
```
Comment (spencerschrock) [inline checker/check_result_test.go:381]: please shadow tt here to avoid race conditions
```
tt := tt
```

File: checker/check_result_test.go [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+			},
+			want: CheckResult{
+				Name:    "name",
+				Reason:  "reason",
+				Score:   10,
+				Version: 2,
+			},
+		},
+	}
+	for _, tt := range tests {
```
Comment (spencerschrock) [inline checker/check_result_test.go:333]: please shadow tt here to avoid race conditions
```
tt := tt
```

File: checker/check_result_test.go [hunk 7]
```diff
@@ -0,0 +1,473 @@
...
+			},
+			want: CheckResult{
+				Name:    "name",
+				Reason:  "reason -- score normalized to 10",
+				Score:   10,
+				Version: 2,
+			},
+		},
+	}
+	for _, tt := range tests {
```
Comment (spencerschrock) [inline checker/check_result_test.go:285]: please shadow tt here to avoid race conditions
```
tt := tt
```

File: checker/check_result_test.go [hunk 8]
```diff
@@ -0,0 +1,473 @@
...
+			},
+			want: CheckResult{
+				Name:    "name",
+				Reason:  "reason",
+				Version: 2,
+				Score:   1,
+			},
+		},
+	}
+	for _, tt := range tests {
```
Comment (spencerschrock) [inline checker/check_result_test.go:231]: please shadow tt here to avoid race conditions
```
tt := tt
```

File: checker/check_result_test.go [hunk 9]
```diff
@@ -0,0 +1,473 @@
...
+			args: args{
+				reason: "reason",
+				score:  1,
+			},
+			want: "reason -- score normalized to 1",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```
Comment (spencerschrock) [inline checker/check_result_test.go:182]: please shadow tt here to avoid race conditions
```
tt := tt
```

File: checker/check_result_test.go [hunk 10]
```diff
@@ -0,0 +1,473 @@
...
+		{
+			name: "multiple",
+			args: args{
+				success: 1,
+				total:   2,
+			},
+			want: 5,
+		},
+	}
+	for _, tt := range tests {
```
Comment (spencerschrock) [inline checker/check_result_test.go:143]: please shadow tt here to avoid race conditions
```
tt := tt
```

File: checker/check_result_test.go [hunk 11]
```diff
@@ -0,0 +1,473 @@
...
+		},
+		{
+			name: "multiple",
+			args: args{
+				scores: map[int]int{1: 1, 2: 2, 3: 3},
+			},
+			want: 2,
+		},
+	}
+	for _, tt := range tests {
```
Comment (spencerschrock) [inline checker/check_result_test.go:98]: please shadow tt here to avoid race conditions
```
tt := tt
```

File: checker/check_result_test.go [hunk 12]
```diff
@@ -0,0 +1,473 @@
...
+		},
+		{
+			name: "multiple",
+			args: args{
+				scores: []int{1, 2, 3},
+			},
+			want: 2,
+		},
+	}
+	for _, tt := range tests {
```
Comment (spencerschrock) [inline checker/check_result_test.go:56]: please shadow tt here to avoid race conditions
```
tt := tt
```

File: checker/check_request_test.go [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+		{
+			name: "empty exists",
+			args: args{
+				in:     []RequestType{FileBased, CommitBased},
+				exists: CommitBased,
+			},
+			want: true,
+		},
+	}
+	for _, tt := range tests {
```
Comment (spencerschrock) [inline checker/check_request_test.go:112]: please shadow tt here to avoid race conditions
```
tt := tt
```

File: .golangci.yml (hunk 2 above)
Comment (naveensrinivasan) [reply .golangci.yml]: Agreed. Thanks

File: checker/check_result_test.go (hunk 3 above)
Comment (naveensrinivasan) [reply checker/check_result_test.go:475]: Resolved all of them. Not commenting on each one of them.

**Pull request merged**